sentry/src/main/java/io/sentry/Hub.java

```diff
@@ -593,6 +595,26 @@ public void withScope(final @NotNull ScopeCallback callback) {
     }
   }
 
+  @Override
+  public void withIsolationScope(final @NotNull ScopeCallback callback) {
+    if (!isEnabled()) {
+      try {
+        callback.run(NoOpScope.getInstance());
+      } catch (Throwable e) {
+        options.getLogger().log(SentryLevel.ERROR, "Error in the 'withScope' callback.", e);
+      }
+
+    } else {
+      pushScope();
+      try {
+        callback.run(stack.peek().getScope());
+      } catch (Throwable e) {
+        options.getLogger().log(SentryLevel.ERROR, "Error in the 'withScope' callback.", e);
+      }
+      popScope();
```

Comment:
wouldn't it be better to use the token from `pushScope();` instead and call close on it?

Reply:
You mean on fork of `Scopes` we add the lifecycle token to the new `Scopes` instance where we weakly reference it to be able to then pop it?

I'll have to think about this some more but could be a good solution for the hack with getParent().makeCurrent() that's currently in place.